pkg/functions/function_test.go

```diff
@@ -220,3 +226,176 @@ func TestFunction_MigrationError(t *testing.T) {
+	}
+	if !f.Built() {
+		t.Fatal("freshly built function reported Built==false (3)")
+	}
+
+	// Remove the testfile, which should result in the client reporting that
+	// the function is no longer Built (stale)
+	if err := os.Remove(filepath.Join(root, testfile)); err != nil {
+		t.Fatal(err)
+	}
+	if f.Built() {
+		t.Fatal("client did not detect a removed file as indicating build staleness")
+	}
+}
+
+// TestFunction_Stamp ensures that the Stamp method and it's associated
+// accessor BuildStamp:
+//
+//		yields an empty string if the function is unbuilt
+//		yields a build stamp once built
+//		The value is unchanged on multiple invocations with an unchanged fs.
+//		The value changes if the filesystem changes.
+//	 Creates a journal when requested.
+func TestFunction_Stamp(t *testing.T) {
+	root, rm := Mktemp(t)
+	defer rm()
+
+	f := fn.Function{Root: root, Runtime: "go", Name: "f"}
+	client := fn.New(fn.WithBuilder(mock.NewBuilder()), fn.WithRegistry(TestRegistry))
+	stamp := f.BuildStamp()
+
+	// In-memory functions should have no buildstamp
+	if stamp != "" {
+		t.Fatalf("build stamp of an uninitialized function should be '', got '%v'", stamp)
+	}
+
+	// Initialized (but not built) functions should also have no stamp
+	f, err := client.Init(f)
+	if err != nil {
+		t.Fatal(err)
+	}
+	stamp = f.BuildStamp()
+	if stamp != "" {
+		t.Fatalf("initial build stamp of an unbuilt but initialized function should be empty, got '%v'", stamp)
+	}
+
+	// Built functions should have a stamp
+	f, err = client.Build(context.Background(), f)
+	if err != nil {
+		t.Fatal(err)
+	}
+	stamp = f.BuildStamp()
+	if stamp == "" {
+		t.Fatal("building the function did not yield a build stamp")
+	}
+
+	// Explicitly stamping again should have no effect
+	if err = f.Stamp(); err != nil {
+		t.Fatal(err)
+	}
+	stamp2 := f.BuildStamp()
+	if stamp2 != stamp {
+		t.Fatalf("re-stamping an unchanged function changed its stamp.  expected '%v', got '%v'", stamp, stamp2)
+	}
+
+	// Windows is randomly failing the following test.  This is a quick
+	// way to confirm it's a racing condition with fs modification.
+	// Test succeeds reliably on linux, and there is an explicit .Flush
+	time.Sleep(1 * time.Second)
+
+	// Editing the filesystem and re-stamping should have an effect
+	if err := os.Chtimes(filepath.Join(root, "func.yaml"), time.Now(), time.Now()); err != nil {
+		fmt.Println(err)
+	}
+	if err = f.Stamp(); err != nil {
+		t.Fatal(err)
+	}
+	stamp2 = f.BuildStamp()
+	if stamp2 == "" {
+		t.Fatal("stamping a built function which has had disk changes since build resulted in an empty stamp.")
+	}
+	if stamp2 == stamp {
+		t.Fatalf("stamping a changed function did not change stamp.  got '%v' again", stamp2)
+	}
+
+	// Asking to stamp again with a journal should result in there being
+	// a "[timestamp]built.log" file in .func
+	if err = f.Stamp(fn.WithStampJournal()); err != nil {
+		t.Fatal(err)
+	}
+	files, err := ioutil.ReadDir(filepath.Join(root, fn.RunDataDir))
```

Comment:
Change this to `os.ReadDir()` to fix the linter error, please

Reply:
I should have noticed that